eel/eel-gtk-extensions.c

```diff
@@ -169,8 +184,23 @@ eel_gtk_window_set_initial_geometry (GtkWindow *window,
         real_top = top;
 
         screen = gtk_window_get_screen (window);
-        screen_width  = gdk_screen_get_width  (screen);
-        screen_height = gdk_screen_get_height (screen);
+        display = gdk_screen_get_display (screen);
+
+        if  (GDK_IS_X11_DISPLAY (display))
+        {
+            scale = gtk_widget_get_scale_factor (GTK_WIDGET (window));
+            screen_width  = WidthOfScreen (gdk_x11_screen_get_xscreen (screen)) / scale;
+            screen_height = HeightOfScreen (gdk_x11_screen_get_xscreen (screen)) / scale;
+        }
+        else
+        {
+            GdkRectangle geometry = {0};
+            GdkMonitor *monitor;
+            monitor = gdk_display_get_monitor (display, 0);
+            gdk_monitor_get_geometry(monitor, &geometry);
+            screen_width = geometry.width;
+            screen_height = geometry.height;
+        }
```

Comment:
Before this change we have 2 lines of (well working) code.
I do not get why this should blow up, only because of an idea to use at all places the same old xlib macros.
Do we have crashes in x11 at this place?
Does anyone have a stacktrace which point to this 2 lines?
I recall in 2019 when all this xlib macros were added by a deprecation fix PR, that we get warned by a gnome dev not to do this, because it is only usable for x11.
I won't review it because i am thinking the idea of this PR is wrong.
Anyway, i don't block it when other devs from @mate-desktop/core-team  think this is an excellent idea.